flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/debezium/task/MySqlSnapshotSplitReadTask.java

```diff
@@ -313,7 +316,14 @@ else if (actualColumn.jdbcType() == Types.TINYINT
         else if ("CHAR".equals(actualColumn.typeName())
                 || "VARCHAR".equals(actualColumn.typeName())
                 || "TEXT".equals(actualColumn.typeName())) {
-            return rs.getBytes(fieldNo);
+            Charset columnCharset = DebeziumUtils.charsetFor(actualColumn);
+            final byte[] stringColumnBytes = rs.getBytes(fieldNo);
+            if (stringColumnBytes != null && columnCharset != null) {
+                // UTF_8 for the jdbc connection charset encoding
+                return new String(stringColumnBytes, StandardCharsets.UTF_8)
```

Comment:
I find currently users can set the jdbc parameters in the with option `jdbc.properties.*`[1]. Therefore, the charset encoding may not  UTF_8 here.

I just wonder whether we should use the default encoding format here.  The parameter `characterSetResults` means return the result with the default value if no set. Could you check what's the jdbc url in the debezium?

[1] https://ververica.github.io/flink-cdc-connectors/master/content/connectors/mysql-cdc.html#connector-options
[2] https://docs.oracle.com/cd/E17952_01/connector-j-8.0-en/connector-j-connp-props-session.html#cj-conn-prop_characterSetResults

Reply:
in the debezium class MysqlConnection, the jdbc url pattern is : (characterEncoding=UTF-8&characterSetResults=UTF-8)
```
protected static final String URL_PATTERN = "jdbc:mysql://${hostname}:${port}/?useInformationSchema=true&nullCatalogMeansCurrent=false&useSSL=${useSSL}&useUnicode=true&characterEncoding=UTF-8&characterSetResults=UTF-8&zeroDateTimeBehavior=CONVERT_TO_NULL&connectTimeout=${connectTimeout}";
```

So I think it is ok to use the default UTF-8.